src/components/User/User.js

```diff
@@ -15,15 +17,33 @@ const User = ({ user }) => {
           <li>Country: {country}</li>
           <li>Company: {company}</li>
         </ul>
-        <div className="User__about">
+        {isExpanded && <div className="User__about">
           <h3>About {name.split(' ')[0]}:</h3>
           <p>{about}</p>
-        </div>
+        </div>}
       </div>
       <div className="User__controls">
-        <button>click me</button>
+        <button onClick={handleToggleExpanded}>
+          {isExpanded ? "show less" : "show more"}
+        </button>
       </div>
-    </section>
+    </section> :
+    /* Not sure if I misunderstood this requirement:
+    Implement the SearchBar functionality.  
+    When the user types in the search bar, 
+    show only the users whose name, country, or company match the user input. 
+    That's why I added this ternary.
+    */ 
```

Comment:
Thank you for adding this comment. I think you can implement this without passing `searchInput` down through to the user. I might have missed something, but I think your code in `<App />` already takes care of filtering. If `searchInput` is falsey (empty string), I believe all of the student cards will render anyway.

Reply:
I read this requirement again, I thought I misunderstood it before. Yes, it makes sense that I don't need the big ternary. I thought we only need to show these three information after filtering but it was wrong.